optuna/samplers/base.py

```diff
@@ -120,3 +120,16 @@ def sample_independent(self, study, trial, param_name, param_distribution):
         """
 
         raise NotImplementedError
+
+    @abc.abstractmethod
+    def reseed_rng(self) -> None:
+        """Reseed sampler's random number generator.
+
+        This method is called by the :class:`~optuna.study.Study` instance if trials are executed
+        in parallel with the option ``n_jobs>1``. In that case, the sampler instance will be
+        replicated including the state of the random number generator, and they may suggest the
+        same values. To prevent this issue, this method assigns a different seed to each random
+        number generator.
+        """
+
+        raise NotImplementedError
```

Comment:
This always requires an additional step of overriding the method when the user want to make a new sampler, which I think a bit burdensome. Can't we just do `pass` instead of `NotImplementedSampler`?

Reply:
Thank you for your suggestion. That makes sense.
The current implementation will not work with existing user-defined samplers.
I fixed it in https://github.com/optuna/optuna/pull/968/commits/bcdcf9d2944d67db42546f9b1b6f42e8c512dcb5.